internal/proxy/hook/authz/authz.go

```diff
@@ -71,8 +63,10 @@ func (a Authz) ServeHook(res *http.Response, err error) (*http.Response, error)
 		return a.escape.ServeHook(res, err)
 	}
 
-	ruleFromRequest, ok := hook.ExtractRule(res.Request)
-	if !ok {
+	attributes := res.Request.Context().Value("requestAttributes").(map[string]any)
```

Comment:
this can cause panics if the type changes or the value is not set etc. 

Also, using primitive Go types as context-keys is highly discouraged. Ideally, the package which is responsible for injecting this particular value into context should also expose a function to extract the value. The details about what key is used, what is the type of the key, etc. should be hidden awy.

```golang
package requestctx

var reqAttribsKey = ctxKey("request_attributes")

type ctxKey string // should be private.

func InjectAttribs(ctx context.Context, value map[string]any) context.Context {
   return context.WithValue(ctx, reqAttribsKey, value)
}

func AttribsFrom(ctx context.Context) map[string]any {
   // even if the second value is ignored, this operation is safe. 
   // using "v := ctx.Value(reqAttribsKey).(map[string]any)" is NOT safe.
   v, _ := ctx.Value(reqAttribsKey).(map[string]any)
   return v
}
```